### PR TITLE
VB-574: Date/time picker - form fixes and pre-populate from session

### PR DESCRIPTION
**Pull request by tpmcgowan**

### CODE REVIEW

File: server/routes/visitors.test.ts [hunk 1]
```diff
@@ -2,29 +2,20 @@ import type { Express } from 'express'
...
+
 let sessionApp: Express
-let prisonerVisitorsService: PrisonerVisitorsService
 const systemToken = async (user: string): Promise<string> => `${user}-token-1`
 
 let flashData: Record<'errors' | 'formValues', Record<string, string | string[]>[]>
-let returnData: { prisonerName: string; visitorList: VisitorListItem[] }
 let visitSessionData: VisitSessionData
 
-class MockPrisonerVisitorsService extends PrisonerVisitorsService {
```
Comment (tpmcgowan) [inline server/routes/visitors.test.ts:17]: Changed the way services mocked as adding a second service caused lint errors about multiple class definitions in one file...

File: server/routes/visitors.test.ts [hunk 2]
```diff
@@ -465,6 +487,290 @@ describe('POST /visit/select-visitors', () => {
   })
 })
 
+describe('/visit/select-date-and-time', () => {
```
Comment (tpmcgowan) [inline server/routes/visitors.test.ts:490]: Wrapped these in another level of `describe` and shuffled some of the test variables around a little to try and keep stuff closer to where it's needed - hopefully making file easier to eventually split up.

File: server/views/pages/dateAndTime.njk [hunk 3]
```diff
@@ -195,10 +193,15 @@
       }) }}
     {% endfor %}
     
-        {{ govukButton({
-          text: "Continue"
-        }) }} 
+    {{ govukButton({
```
Comment (tpmcgowan) [inline server/views/pages/dateAndTime.njk:196]: Moved this around so Continue button doesn't show if there are no slots to choose.

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**